cmd/backup-manager/app/util/util.go

```diff
@@ -161,70 +161,103 @@ func GetOptionValueFromEnv(option, envPrefix string) string {
 // ConstructBRGlobalOptionsForBackup constructs BR global options for backup and also return the remote path.
 func ConstructBRGlobalOptionsForBackup(backup *v1alpha1.Backup) ([]string, error) {
 	var args []string
-	config := backup.Spec.BR
-	if config == nil {
+	config := backup.Spec
+	if config.BR == nil {
 		return nil, fmt.Errorf("no config for br in backup %s/%s", backup.Namespace, backup.Name)
 	}
-	args = append(args, constructBRGlobalOptions(config)...)
+	args = append(args, constructBRGlobalOptions(config.BR)...)
 	storageArgs, err := getRemoteStorage(backup.Spec.StorageProvider)
 	if err != nil {
 		return nil, err
 	}
 	args = append(args, storageArgs...)
-	if (backup.Spec.Type == v1alpha1.BackupTypeDB || backup.Spec.Type == v1alpha1.BackupTypeTable) && config.DB != "" {
-		args = append(args, fmt.Sprintf("--db=%s", config.DB))
-	}
-	if backup.Spec.Type == v1alpha1.BackupTypeTable && config.Table != "" {
-		args = append(args, fmt.Sprintf("--table=%s", config.Table))
+
+	if config.TableFilter != nil && len(config.TableFilter) > 0 {
+		for _, tableFilter := range config.TableFilter {
+			args = append(args, "--filter", tableFilter)
+		}
+	} else {
+		if ((backup.Spec.Type == v1alpha1.BackupTypeDB || backup.Spec.Type == v1alpha1.BackupTypeTable) && config.BR.DB != "") ||
+			(backup.Spec.Type == v1alpha1.BackupTypeTable && config.BR.Table != "") {
+			if (backup.Spec.Type == v1alpha1.BackupTypeDB || backup.Spec.Type == v1alpha1.BackupTypeTable) && config.BR.DB != "" {
+				args = append(args, fmt.Sprintf("--db=%s", config.BR.DB))
+			}
+			if backup.Spec.Type == v1alpha1.BackupTypeTable && config.BR.Table != "" {
+				args = append(args, fmt.Sprintf("--table=%s", config.BR.Table))
+			}
+		} else {
+			args = append(args, defaultTableFilterOptions...)
+		}
 	}
+
 	return args, nil
```

Comment:
How about:
```go
if config.TableFilter != nil && len(config.TableFilter) > 0 {
    for _, tableFilter := range config.TableFilter {
        args = append(args, "--filter", tableFilter)
    }
    return args, nil
}

switch backup.Spec.Type {
case v1alpha1.BackupTypeTable:
    if config.BR.Table != "" {
        args = append(args, fmt.Sprintf("--table=%s", config.BR.Table))
    }
    if config.BR.DB != "" {
        args = append(args, fmt.Sprintf("--db=%s", config.BR.DB))
    }
    if config.BR.Table == "" && config.BR.DB == "" {
        args = append(args, defaultTableFilterOptions...)
    }
case v1alpha1.BackupTypeDB:
    if config.BR.DB != "" {
        args = append(args, fmt.Sprintf("--db=%s", config.BR.DB))
    } else {
        args = append(args, defaultTableFilterOptions...)
    }
}
```
I think it will be clear and understandable.

Reply:
Thanks this is much better than what I did. I will test its working manually. Then commit it and do the tests for BR.

Reply:
No need to add `defaultTableFilterOptions` for BR, this will change the previous backup behavior of BR.